src/period-calculation/generate-fixed-periods/generate-fixed-periods.ts

```diff
@@ -19,11 +19,21 @@ type GenerateFixedPeriods = (options: {
     locale?: string
     startingDay?: number /** 1 is Monday */
     yearsCount?: number | null
+    endsBefore?: string
 }) => Array<FixedPeriod>
 
+/**
+ * @param {Object} options
+ * @param {string} [options.endsBefore] - A date as string to exclude all
+ * periods that do not end before the provided date. This will help generating
+ * periods up to a certain point (e.g. "now", in two weeks when period type is
+ * WEEKLY and two open future periods are allowed, like in the aggregate data
+ * entry app)
+ */
```

Comment:
I'm not really sold on this option property name. Basically, you're imposing a cut-off date, right? So anything ending later than `endsBefore` is excluded?
Perhaps it's more the description that is confusing rather than the property name...
> A date as string to exclude all periods that do not end before the provided date

This sentence basically has a double negation, in the form of _exclude_ and _do not_. Rewriting that to
> A date as string to only include periods which end before the provided date

Would already help a lot. But perhaps using a different property name, such as `maxPeriodEndDate` would help some more.